store.go

```diff
@@ -3843,3 +3853,41 @@ func (s *store) MultiList(options MultiListOptions) (MultiListResult, error) {
 	}
 	return out, nil
 }
+
+// Dedup deduplicates layers in the store.
+func (s *store) Dedup(req DedupArgs) (drivers.DedupResult, error) {
+	imgs, err := s.Images()
+	if err != nil {
+		return drivers.DedupResult{}, err
+	}
+	var topLayers []string
+	for _, i := range imgs {
+		topLayers = append(topLayers, i.TopLayer)
+		topLayers = append(topLayers, i.MappedTopLayers...)
+	}
+	return writeToLayerStore(s, func(rlstore rwLayerStore) (drivers.DedupResult, error) {
+		layers := make(map[string]interface{})
+		for _, i := range topLayers {
+			cur := i
+			for cur != "" {
+				layers[cur] = nil
```

Comment:
(Considering that this code is already handling `ErrLayerUnknown`, it would be a tiny bit cleaner to only add a layer to `layers` after the `.Get()` below succeeds. Ultimately, the two implemented graph drivers filter out unknown layers anyway, but doing it centrally here at the top level would cost us nothing.)